src/export/StructureDefinitionExporter.ts

```diff
@@ -414,9 +414,9 @@ export class StructureDefinitionExporter implements Fishable {
         // The FHIR spec prohibits constraining any parent element in a 'specialization'
         // (i.e., logical model and resource), therefore log an error if that is attempted
         // and continue to the next rule.
-        if (element.path !== element.base.path) {
-          // The AddElementRule always sets the newElementBase.path to the value of element.path.
-          // All parent elements will have the element.base.path pointing to the parent.
+        const baseElementPath = element.path.replace(`${element.structDef.id}.`, '');
+        const addElementRules = fshDefinition.rules.filter(rule => rule instanceof AddElementRule);
```

Comment:
This seems like kind of an expensive operation to do for every iteration through the loop.  Can we put it outside the loop?